vllm/sampling_params.py

```diff
@@ -148,7 +148,14 @@ def __init__(
         self.prompt_logprobs = prompt_logprobs
         self.skip_special_tokens = skip_special_tokens
         self.spaces_between_special_tokens = spaces_between_special_tokens
-        self.logits_processors = logits_processors
+        # A separate logit processor is needed for each output sequence
+        # since certain logits processors (such as BaseGuidedLogitsProcessor)
+        # in multi-beam generation must track the sequences generated
+        # by each beam up to that point.
+        # See https://github.com/vllm-project/vllm/issues/3448 for more
+        self.logits_processors = [
+            copy.deepcopy(logits_processors) for _ in range(n)
```

Comment:
same as comments above, i believe we don't need deepcopy here, simple copy might work

Reply:
Here it is actually needed, since every `logits_processors` needs to maintain independent state in multi-beam setting. I checked if it works with `copy`, but unfortunately it failed